dotnet/src/InternalUtilities/src/System/TypeConverterFactory.cs

```diff
@@ -51,6 +53,21 @@ internal static class TypeConverterFactory
             return converter;
         }
 
-        return null;
+        return new JsonSerializationTypeConverter();
+    }
+
+    internal sealed class JsonSerializationTypeConverter : TypeConverter
+    {
+        public override bool CanConvertFrom(ITypeDescriptorContext? context, Type sourceType) => true;
+
+        public override object? ConvertFrom(ITypeDescriptorContext? context, CultureInfo? culture, object value)
+        {
+            return JsonSerializer.Deserialize<object>((string)value);
+        }
+
+        public override object? ConvertTo(ITypeDescriptorContext? context, CultureInfo? culture, object? value, Type destinationType)
+        {
+            return JsonSerializer.Serialize(value);
```

Comment:
Both ConvertFrom and ConvertTo here are making assumptions about the source and destination types. This is going to require some re-work as part of @SergeyMenshykh's work to support marshaling with types other than string.

Reply:
I've setup a sync with him for the morning to get on the same page about this.

Reply:
resolving in favor of the other comment